### ISSUE TITLE
[QA] Reconnect button is enabled if the account is logged out

### ISSUE THREAD

**Issue by HanaGemela** [owncloud/client#10407]
### Pre-submission Checks

- [X] I [checked for similar issues](https://github.com/owncloud/client/issues?q=), but could not find any. I also checked the closed issues. I could not contribute additional information to any existing issue.
- [X] I will take the time to fill in all the required fields. I know that the bug report may be dismissed otherwise due to lack of information.

### Describe the QA issue

Reconnect button doesn't react.
ownCloud 3.1.0.9804-rc1 
mac 13.0 
ocis 2.0.0

### Steps to reproduce the issue

1. Sign out
2. Reconnect

### Screenshots

No screenshot

### Expected behavior

The button either behaves like Log in button or is not available (is greyed out)

### Actual behavior

Button is available but doesn't do anything

**Comment by TheOneRing** [comment]
Please provide logs.

**Comment by HanaGemela** [comment]
```
23-01-18 13:28:03:482 [ info sync.httplogger ]:	"02d4d90e-4547-4c2c-a20a-d04e1625706a: Response: PROPFIND 207 (14ms) https://ocis.owncloud.test/dav/spaces/4808fe4d-d318-4038-b5ba-42db0de52cbd$ea0c6569-7c38-4333-974a-174572c96f25/ Header: { Access-Control-Allow-Origin: *, Access-Control-Expose-Headers: Tus-Resumable, Tus-Version, Tus-Extension, Content-Length: 376, Content-Security-Policy: default-src 'none';, Content-Type: application/xml; charset=utf-8, Date: Wed, 18 Jan 2023 12:28:03 GMT, Dav: 1, 3, extended-mkcol, Preference-Applied: return=minimal, Tus-Extension: creation,creation-with-upload,checksum,expiration, Tus-Resumable: 1.0.0, Tus-Version: 1.0.0, Vary: Prefer, X-Content-Type-Options: nosniff, X-Download-Options: noopen, X-Frame-Options: SAMEORIGIN, X-Permitted-Cross-Domain-Policies: none, X-Robots-Tag: none, X-Xss-Protection: 1; mode=block, } Data: [<d:multistatus xmlns:s=\"http://sabredav.org/ns\" xmlns:d=\"DAV:\" xmlns:oc=\"http://owncloud.org/ns\"><d:response><d:href>/dav/spaces/4808fe4d-d318-4038-b5ba-42db0de52cbd$ea0c6569-7c38-4333-974a-174572c96f25/</d:href><d:propstat><d:prop><d:getetag>\"bd76ecf45051b9481dda3e48f100af55\"</d:getetag></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response></d:multistatus>]"
23-01-18 13:28:03:482 [ info sync.networkjob.propfind ]:	LSCOL of QUrl("https://ocis.owncloud.test/dav/spaces/4808fe4d-d318-4038-b5ba-42db0de52cbd$ea0c6569-7c38-4333-974a-174572c96f25/") FINISHED WITH STATUS "OK"
23-01-18 13:28:03:482 [ info gui.folder.manager ]:	void OCC::FolderMan::setSyncEnabled(bool) true
23-01-18 13:28:03:483 [ debug sync.networkjob ]	[ OCC::AbstractNetworkJob::slotFinished ]:	Network job finished OCC::RequestEtagJob(OCC::Account("lucy@ocis.owncloud.test"), "https://ocis.owncloud.test/dav/spaces/4808fe4d-d318-4038-b5ba-42db0de52cbd$ea0c6569-7c38-4333-974a-174572c96f25/", "PROPFIND", Original-Request-ID: "02d4d90e-4547-4c2c-a20a-d04e1625706a", X-Request-ID: "02d4d90e-4547-4c2c-a20a-d04e1625706a")
23-01-18 13:28:03:485 [ info sync.httplogger ]:	"a46fbd9f-0da0-42af-a68d-4840afe79785: Response: GET 200 (17ms) https://ocis.owncloud.test/status.php Header: { Access-Control-Allow-Origin: *, Content-Length: 269, Content-Security-Policy: default-src 'none';, Content-Type: application/json, Date: Wed, 18 Jan 2023 12:28:03 GMT, X-Content-Type-Options: nosniff, X-Download-Options: noopen, X-Frame-Options: SAMEORIGIN, X-Permitted-Cross-Domain-Policies: none, X-Robots-Tag: none, X-Xss-Protection: 1; mode=block, } Data: [{\n    \"installed\": true,\n    \"maintenance\": false,\n    \"needsDbUpgrade\": false,\n    \"version\": \"10.11.0.0\",\n    \"versionstring\": \"10.11.0\",\n    \"edition\": \"Community\",\n    \"productname\": \"Infinite Scale\",\n    \"product\": \"Infinite Scale\",\n    \"productversion\": \"2.0.0\"\n}]"
23-01-18 13:28:03:485 [ info sync.checkserverjob ]:	status.php returns:  QJsonDocument({"edition":"Community","installed":true,"maintenance":false,"needsDbUpgrade":false,"product":"Infinite Scale","productname":"Infinite Scale","productversion":"2.0.0","version":"10.11.0.0","versionstring":"10.11.0"})   QNetworkReply::NoError  Reply:  QNetworkReplyHttpImpl(0x600001b93670)
23-01-18 13:28:03:485 [ info sync.connectionvalidator ]:	** Application: ownCloud found:  QUrl("https://ocis.owncloud.test/")  with version  "10.11.0"
23-01-18 13:28:03:485 [ debug sync.connectionvalidator ]	[ OCC::ConnectionValidator::checkAuthentication ]:	# Check whether authenticated propfind works.
23-01-18 13:28:03:486 [ info sync.httplogger ]:	"6a4f2b33-43a4-4d84-8fd3-b5dd0a4ec8c6: Request: PROPFIND https://ocis.owncloud.test/remote.php/webdav/ Header: { Depth: 0, Prefer: return=minimal, Authorization: Bearer [redacted], User-Agent: Mozilla/5.0 (Macintosh) mirall/3.1.0.9804-rc1 (ownCloud, osx-22.1.0 ClientArchitecture: x86_64 OsArchitecture: x86_64), Accept: */*, Content-Type: text/xml; charset=utf-8, X-Request-ID: 6a4f2b33-43a4-4d84-8fd3-b5dd0a4ec8c6, Original-Request-ID: 6a4f2b33-43a4-4d84-8fd3-b5dd0a4ec8c6, Content-Length: 116, } Data: [<?xml version=\"1.0\" encoding=\"utf-8\"?><d:propfind xmlns:d=\"DAV:\"><d:prop><d:getlastmodified/></d:prop></d:propfind>\n]"
23-01-18 13:28:03:486 [ info sync.networkjob ]:	Created OCC::PropfindJob(OCC::Account("lucy@ocis.owncloud.test"), "https://ocis.owncloud.test/remote.php/webdav/", "PROPFIND", Original-Request-ID: "6a4f2b33-43a4-4d84-8fd3-b5dd0a4ec8c6", X-Request-ID: "6a4f2b33-43a4-4d84-8fd3-b5dd0a4ec8c6") for OCC::ConnectionValidator(0x6000017edc20)
23-01-18 13:28:03:504 [ info sync.httplogger ]:	"6a4f2b33-43a4-4d84-8fd3-b5dd0a4ec8c6: Response: PROPFIND 207 (18ms) https://ocis.owncloud.test/remote.php/webdav/ Header: { Access-Control-Allow-Origin: *, Access-Control-Expose-Headers: Tus-Resumable, Tus-Version, Tus-Extension, Content-Length: 320, Content-Security-Policy: default-src 'none';, Content-Type: application/xml; charset=utf-8, Date: Wed, 18 Jan 2023 12:28:03 GMT, Dav: 1, 3, extended-mkcol, Preference-Applied: return=minimal, Tus-Extension: creation,creation-with-upload,checksum,expiration, Tus-Resumable: 1.0.0, Tus-Version: 1.0.0, Vary: Prefer, X-Content-Type-Options: nosniff, X-Download-Options: noopen, X-Frame-Options: SAMEORIGIN, X-Permitted-Cross-Domain-Policies: none, X-Robots-Tag: none, X-Xss-Protection: 1; mode=block, } Data: [<d:multistatus xmlns:s=\"http://sabredav.org/ns\" xmlns:d=\"DAV:\" xmlns:oc=\"http://owncloud.org/ns\"><d:response><d:href>/remote.php/webdav/</d:href><d:propstat><d:prop><d:getlastmodified>Wed, 18 Jan 2023 12:24:24 GMT</d:getlastmodified></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response></d:multistatus>]"
23-01-18 13:28:03:504 [ info sync.networkjob.propfind ]:	LSCOL of QUrl("https://ocis.owncloud.test/remote.php/webdav/") FINISHED WITH STATUS "OK"
23-01-18 13:28:03:504 [ info sync.httplogger ]:	"84d3e081-963a-49a4-8697-4f7f9da4e4cb: Request: GET https://ocis.owncloud.test/ocs/v2.php/cloud/capabilities?format=json Header: { OCS-APIREQUEST: true, Authorization: Bearer [redacted], User-Agent: Mozilla/5.0 (Macintosh) mirall/3.1.0.9804-rc1 (ownCloud, osx-22.1.0 ClientArchitecture: x86_64 OsArchitecture: x86_64), Accept: */*, X-Request-ID: 84d3e081-963a-49a4-8697-4f7f9da4e4cb, Original-Request-ID: 84d3e081-963a-49a4-8697-4f7f9da4e4cb, } Data: []"
23-01-18 13:28:03:504 [ info sync.networkjob ]:	Created OCC::JsonApiJob(OCC::Account("lucy@ocis.owncloud.test"), "https://ocis.owncloud.test/ocs/v2.php/cloud/capabilities?format=json", "GET", Original-Request-ID: "84d3e081-963a-49a4-8697-4f7f9da4e4cb", X-Request-ID: "84d3e081-963a-49a4-8697-4f7f9da4e4cb") for OCC::ConnectionValidator(0x6000017edc20)
23-01-18 13:28:03:504 [ debug sync.networkjob ]	[ OCC::AbstractNetworkJob::slotFinished ]:	Network job finished OCC::PropfindJob(OCC::Account("lucy@ocis.owncloud.test"), "https://ocis.owncloud.test/remote.php/webdav/", "PROPFIND", Original-Request-ID: "6a4f2b33-43a4-4d84-8fd3-b5dd0a4ec8c6", X-Request-ID: "6a4f2b33-43a4-4d84-8fd3-b5dd0a4ec8c6")
23-01-18 13:28:03:513 [ info sync.httplogger ]:	"84d3e081-963a-49a4-8697-4f7f9da4e4cb: Response: GET 200 (9ms) https://ocis.owncloud.test/ocs/v2.php/cloud/capabilities?format=json Header: { Content-Type: application/json; charset=utf-8, Date: Wed, 18 Jan 2023 12:28:03 GMT, Ocs-Api-Version: 2, Vary: Origin, Transfer-Encoding: chunked, } Data: [{\"ocs\":{\"meta\":{\"status\":\"ok\",\"statuscode\":200,\"message\":\"OK\"},\"data\":{\"capabilities\":{\"core\":{\"pollinterval\":60,\"webdav-root\":\"remote.php/webdav\",\"status\":{\"installed\":true,\"maintenance\":false,\"needsDbUpgrade\":false,\"version\":\"10.11.0.0\",\"versionstring\":\"10.11.0\",\"edition\":\"Community\",\"productname\":\"Infinite Scale\",\"product\":\"Infinite Scale\",\"productversion\":\"2.0.0\"},\"support-url-signing\":true},\"checksums\":{\"supportedTypes\":[\"sha1\",\"md5\",\"adler32\"],\"preferredUploadType\":\"sha1\"},\"files\":{\"privateLinks\":true,\"bigfilechunking\":false,\"undelete\":true,\"versioning\":true,\"favorites\":false,\"blacklisted_files\":[],\"tus_support\":{\"version\":\"1.0.0\",\"resumable\":\"1.0.0\",\"extension\":\"creation,creation-with-upload\",\"max_chunk_size\":100000000,\"http_method_override\":\"\"},\"archivers\":[{\"enabled\":true,\"version\":\"2.0.0\",\"formats\":[\"tar\",\"zip\"],\"archiver_url\":\"/archiver\",\"max_num_files\":\"10000\",\"max_size\":\"1073741824\"}],\"app_providers\":[{\"enabled\":true,\"version\":\"1.1.0\",\"apps_url\":\"/app/list\",\"open_url\":\"/app/open\",\"open_web_url\":\"/app/open-with-web\",\"new_url\":\"/app/new\"}]},\"dav\":{\"chunking\":\"\",\"trashbin\":\"1.0\",\"reports\":[\"search-files\"],\"chunkingParallelUploadDisabled\":false},\"files_sharing\":{\"api_enabled\":true,\"resharing\":true,\"group_sharing\":true,\"sharing_roles\":true,\"deny_access\":false,\"auto_accept_share\":true,\"share_with_group_members_only\":true,\"share_with_membership_groups_only\":true,\"search_min_length\":3,\"default_permissions\":22,\"user_enumeration\":{\"enabled\":true,\"group_members_only\":true},\"federation\":{\"outgoing\":false,\"incoming\":false},\"public\":{\"enabled\":true,\"send_mail\":true,\"social_share\":true,\"upload\":true,\"multiple\":true,\"supports_upload_only\":true,\"password\":{\"enforced_for\":{\"read_only\":false,\"read_write\":false,\"upload_only\":false},\"enforced\":false},\"expire_date\":{\"enabled\":false},\"can_edit\":true,\"alias\":true},\"user\":{\"send_mail\":true,\"profile_picture\":false,\"settings\":[{\"enabled\":true,\"version\":\"1.0.0\"}]}},\"spaces\":{\"version\":\"1.0.0\",\"enabled\":true,\"projects\":true,\"share_jail\":true}},\"version\":{\"major\":10,\"minor\":11,\"micro\":0,\"string\":\"10.11.0\",\"edition\":\"Community\",\"product\":\"Infinite Scale\",\"productversion\":\"2.0.0\"}}}}]"
23-01-18 13:28:03:513 [ info sync.networkjob.jsonapi ]:	JsonJob of QUrl("https://ocis.owncloud.test/ocs/v2.php/cloud/capabilities?format=json") FINISHED WITH STATUS "OK"
23-01-18 13:28:03:514 [ info sync.connectionvalidator ]:	Server capabilities QJsonObject({"checksums":{"preferredUploadType":"sha1","supportedTypes":["sha1","md5","adler32"]},"core":{"pollinterval":60,"status":{"edition":"Community","installed":true,"maintenance":false,"needsDbUpgrade":false,"product":"Infinite Scale","productname":"Infinite Scale","productversion":"2.0.0","version":"10.11.0.0","versionstring":"10.11.0"},"support-url-signing":true,"webdav-root":"remote.php/webdav"},"dav":{"chunking":"","chunkingParallelUploadDisabled":false,"reports":["search-files"],"trashbin":"1.0"},"files":{"app_providers":[{"apps_url":"/app/list","enabled":true,"new_url":"/app/new","open_url":"/app/open","open_web_url":"/app/open-with-web","version":"1.1.0"}],"archivers":[{"archiver_url":"/archiver","enabled":true,"formats":["tar","zip"],"max_num_files":"10000","max_size":"1073741824","version":"2.0.0"}],"bigfilechunking":false,"blacklisted_files":[],"favorites":false,"privateLinks":true,"tus_support":{"extension":"creation,creation-with-upload","http_method_override":"","max_chunk_size":100000000,"resumable":"1.0.0","version":"1.0.0"},"undelete":true,"versioning":true},"files_sharing":{"api_enabled":true,"auto_accept_share":true,"default_permissions":22,"deny_access":false,"federation":{"incoming":false,"outgoing":false},"group_sharing":true,"public":{"alias":true,"can_edit":true,"enabled":true,"expire_date":{"enabled":false},"multiple":true,"password":{"enforced":false,"enforced_for":{"read_only":false,"read_write":false,"upload_only":false}},"send_mail":true,"social_share":true,"supports_upload_only":true,"upload":true},"resharing":true,"search_min_length":3,"share_with_group_members_only":true,"share_with_membership_groups_only":true,"sharing_roles":true,"user":{"profile_picture":false,"send_mail":true,"settings":[{"enabled":true,"version":"1.0.0"}]},"user_enumeration":{"enabled":true,"group_members_only":true}},"spaces":{"enabled":true,"projects":true,"share_jail":true,"version":"1.0.0"}})
23-01-18 13:28:03:514 [ info sync.httplogger ]:	"5b4efa36-6572-4d12-800f-3adf1452618e: Request: GET https://ocis.owncloud.test/ocs/v2.php/cloud/user?format=json Header: { OCS-APIREQUEST: true, Authorization: Bearer [redacted], User-Agent: Mozilla/5.0 (Macintosh) mirall/3.1.0.9804-rc1 (ownCloud, osx-22.1.0 ClientArchitecture: x86_64 OsArchitecture: x86_64), Accept: */*, X-Request-ID: 5b4efa36-6572-4d12-800f-3adf1452618e, Original-Request-ID: 5b4efa36-6572-4d12-800f-3adf1452618e, } Data: []"
23-01-18 13:28:03:514 [ info sync.networkjob ]:	Created OCC::JsonApiJob(OCC::Account("lucy@ocis.owncloud.test"), "https://ocis.owncloud.test/ocs/v2.php/cloud/user?format=json", "GET", Original-Request-ID: "5b4efa36-6572-4d12-800f-3adf1452618e", X-Request-ID: "5b4efa36-6572-4d12-800f-3adf1452618e") for OCC::ConnectionValidator(0x6000017edc20)
23-01-18 13:28:03:514 [ debug sync.networkjob ]	[ OCC::AbstractNetworkJob::slotFinished ]:	Network job finished OCC::JsonApiJob(OCC::Account("lucy@ocis.owncloud.test"), "https://ocis.owncloud.test/ocs/v2.php/cloud/capabilities?format=json", "GET", Original-Request-ID: "84d3e081-963a-49a4-8697-4f7f9da4e4cb", X-Request-ID: "84d3e081-963a-49a4-8697-4f7f9da4e4cb")
23-01-18 13:28:03:523 [ info sync.httplogger ]:	"5b4efa36-6572-4d12-800f-3adf1452618e: Response: GET 200 (9ms) https://ocis.owncloud.test/ocs/v2.php/cloud/user?format=json Header: { Content-Length: 156, Content-Type: application/json; charset=utf-8, Date: Wed, 18 Jan 2023 12:28:03 GMT, Ocs-Api-Version: 2, Vary: Origin, } Data: [{\"ocs\":{\"meta\":{\"status\":\"ok\",\"statuscode\":200,\"message\":\"OK\"},\"data\":{\"id\":\"lucy\",\"display-name\":\"lucy\",\"email\":\"lucy@example.com\",\"user-type\":\"primary\"}}}]"
23-01-18 13:28:03:523 [ info sync.networkjob.jsonapi ]:	JsonJob of QUrl("https://ocis.owncloud.test/ocs/v2.php/cloud/user?format=json") FINISHED WITH STATUS "OK"
23-01-18 13:28:03:523 [ info sync.httplogger ]:	"ce6b1718-5e51-44f9-9a23-bd63daa98fd3: Request: GET https://ocis.owncloud.test/app/list Header: { Authorization: Bearer [redacted], User-Agent: Mozilla/5.0 (Macintosh) mirall/3.1.0.9804-rc1 (ownCloud, osx-22.1.0 ClientArchitecture: x86_64 OsArchitecture: x86_64), Accept: */*, X-Request-ID: ce6b1718-5e51-44f9-9a23-bd63daa98fd3, Original-Request-ID: ce6b1718-5e51-44f9-9a23-bd63daa98fd3, } Data: []"
23-01-18 13:28:03:523 [ info sync.networkjob ]:	Created OCC::JsonJob(OCC::Account("lucy@ocis.owncloud.test"), "https://ocis.owncloud.test/app/list", "GET", Original-Request-ID: "ce6b1718-5e51-44f9-9a23-bd63daa98fd3", X-Request-ID: "ce6b1718-5e51-44f9-9a23-bd63daa98fd3") for QObject(0x0)
23-01-18 13:28:03:523 [ debug sync.networkjob ]	[ OCC::AbstractNetworkJob::slotFinished ]:	Network job finished OCC::JsonApiJob(OCC::Account("lucy@ocis.owncloud.test"), "https://ocis.owncloud.test/ocs/v2.php/cloud/user?format=json", "GET", Original-Request-ID: "5b4efa36-6572-4d12-800f-3adf1452618e", X-Request-ID: "5b4efa36-6572-4d12-800f-3adf1452618e")
23-01-18 13:28:03:527 [ info sync.httplogger ]:	"ce6b1718-5e51-44f9-9a23-bd63daa98fd3: Response: GET 200 (3ms) https://ocis.owncloud.test/app/list Header: { Content-Length: 17, Content-Type: application/json, Date: Wed, 18 Jan 2023 12:28:03 GMT, Vary: Origin, } Data: [{\"mime-types\":[]}]"
23-01-18 13:28:03:527 [ info sync.networkjob.jsonapi ]:	JsonJob of QUrl("https://ocis.owncloud.test/app/list") FINISHED WITH STATUS "OK"
23-01-18 13:28:03:527 [ debug sync.connectionvalidator ]	[ OCC::ConnectionValidator::reportResult ]:	OCC::ConnectionValidator::Connected
23-01-18 13:28:03:527 [ debug sync.networkjob ]	[ OCC::AbstractNetworkJob::slotFinished ]:	Network job finished OCC::JsonJob(OCC::Account("lucy@ocis.owncloud.test"), "https://ocis.owncloud.test/app/list", "GET", Original-Request-ID: "ce6b1718-5e51-44f9-9a23-bd63daa98fd3", X-Request-ID: "ce6b1718-5e51-44f9-9a23-bd63daa98fd3")
23-01-18 13:28:04:841 [ warning default ]:	setActivationPolicy 0 failed
23-01-18 13:28:25:087 [ debug sync.credentials.http ]	[ OCC::HttpCredentials::fetchUser ]:	user already set, no need to fetch from settings
23-01-18 13:28:25:087 [ info sync.account ]:	Clearing cookies
23-01-18 13:28:25:087 [ info sync.credentials.manager ]:	del "ownCloud_credentials:ocis.owncloud.test:99ea89d6-5304-4627-be59-1a4c2c02bb90:http/oauthtoken"
23-01-18 13:28:25:087 [ info sync.account ]:	Clearing cookies
23-01-18 13:28:25:087 [ info gui.account.state ]:	AccountState state change:  "Connected" -> "Signed out"

```

**Comment by TheOneRing** [comment]
Well thx for the log.
The actual issue is that the reconnect button is also active when the account is logged out and not only if we are disconnected.

**Comment by HanaGemela** [comment]
Reconnect button is not available when signed out in ownCloud 3.1.0.9818-rc2 mac
